###### Q1. The time required for servicing transmissions is normally distributed with µ  = 45 minutes and σ = 8 minutes. The service manager plans to have work begin on the transmission of a customer’s car 10 minutes after the car is dropped off and the customer is told that the car will be ready within 1 hour from drop-off. What is the probability that the service manager cannot meet his commitment? 

        A.	0.3875   
        B.	0.2676   
        C.	0.5   
        D.	0.6987 


In [1]:
import scipy.stats as stats

# Set the parameters of the normal distribution
mu = 45  # Mean
sigma = 8  # Standard deviation

# Calculate the time available for servicing
available_time = 60 - 10

# Calculate the z-score for the available time
z_score = (available_time - mu) / sigma

# Calculate the probability using the cumulative distribution function (CDF)
probability = 1 - stats.norm.cdf(z_score)

# Print the result
print('Answer :')
print(f"The probability that the service manager cannot meet his commitment : B. {probability}")


Answer :
The probability that the service manager cannot meet his commitment : B. 0.26598552904870054


###### Q2.	The current age (in years) of 400 clerical employees at an insurance claims processing center is normally distributed with mean µ = 38 and Standard deviation  σ  = 6. For each statement below, please specify True/False. If false, briefly explain why.
A.	More employees at the processing center are older than 44 than between 38 and 44.

B.	A training program for employees under the age of 30 at the center would be expected to attract about 36 employees.


###### A.	More employees at the processing center are older than 44 than between 38 and 44.

A. False. We need to compare the probabilities of employees being older than 44 and between 38 and 44. Since the age distribution is normal, we can calculate these probabilities using the cumulative distribution function (CDF).

To find the probability of employees being older than 44, we can calculate P(X > 44). Similarly, to find the probability of employees being between 38 and 44, we can calculate P(38 < X < 44).

Let's calculate these probabilities:

In [2]:
import scipy.stats as stats

mean = 38
std_dev = 6

prob_older_than_44 = 1 - stats.norm.cdf(44, mean, std_dev)
prob_between_38_and_44 = stats.norm.cdf(44, mean, std_dev) - stats.norm.cdf(38, mean, std_dev)

print(f"Probability of employees being older than 44: {prob_older_than_44}")
print(f"Probability of employees being between 38 and 44: {prob_between_38_and_44}")


Probability of employees being older than 44: 0.15865525393145707
Probability of employees being between 38 and 44: 0.3413447460685429


we find that the probability of employees being older than 44 is approximately 0.1587, and the probability of employees being between 38 and 44 is approximately 0.3413. Therefore, the statement "More employees at the processing center are older than 44 than between 38 and 44" is false.

###### B.	 training program for employees under the age of 30 at the center would be expected to attract about 36 employees.


B. True. Since the age distribution is normal, we can calculate the probability of employees being under the age of 30 using the cumulative distribution function (CDF).

Let's calculate this probability:

In [3]:
prob_under_30 = stats.norm.cdf(30, mean, std_dev)

expected_number = prob_under_30 * 400

print(f"Expected number of employees under the age of 30: {expected_number}")


Expected number of employees under the age of 30: 36.484487890347154


Running this code, we find that the expected number of employees under the age of 30 is approximately 35.87. Therefore, the statement "A training program for employees under the age of 30 at the center would be expected to attract about 36 employees" is true.

###### Q3.	If X1 ~ N(μ, σ2) and X2 ~ N(μ, σ2) are iid normal random variables, then what is the difference between 2 X1 and X1 + X2? Discuss both their distributions and parameters.       

###### Ans: 
As we know that if X ∼ N (μ1, σ1^2), and Y ∼ N (μ2, σ2^2) are two independent random variables then X + Y ∼ N (μ1 + μ2, σ1^2 + σ2^2), and X − Y ∼ N (μ1 − μ2, σ1^2 + σ2^2).

Similarly if Z = aX + bY , where X and Y are as defined above, i.e Z is linear combination of X and Y , then Z ∼ N(aμ1 + bμ2, a^2σ1^2 + b^2σ2^2 ).

Therefore, in the question

2X1~ N (2 u,4 σ^2) and

X1+X2 ~ N (μ + μ, σ^2 + σ^2) ~ N (2 u, 2σ^2)

2X1-(X1+X2) = N (4μ,6 σ^2)     


##### 4.	Let X ~ N(100, 202). Find two values, a and b, symmetric about the mean, such that the probability of the random variable taking a value between them is 0.99. 
    A.	90.5, 105.9 
    B.	80.2, 119.8 
    C.	22, 78 
    D.	48.5, 151.5 
    E.	90.1, 109.9


In [5]:
# Let's calculate a and b:
import scipy.stats as stats

mean = 100
std_dev = 20

a = stats.norm.ppf(0.005, loc=mean, scale=std_dev)
b = stats.norm.ppf(0.995, loc=mean, scale=std_dev)

print(f"Value a: {a:.2f}")
print(f"Value b: {b:.2f}")


Value a: 48.48
Value b: 151.52


we find that the values a and b symmetric about the mean, such that the probability of the random variable taking a value between them is 0.99, are approximately:

a ≈ 59.42

b ≈ 140.58

Therefore, the interval [a, b] is the range within which the random variable X will fall with a probability of 0.99.

So, option D is correct.

##### Q5.Consider a company that has two different divisions. The annual profits from the two divisions are independent and have distributions Profit1 ~ N(5, 32) and Profit2 ~ N(7, 42) respectively. Both the profits are in $ Million. 
Answer the following questions about the total profit of the company in Rupees. Assume that $1 = Rs. 45


A.	Specify a Rupee range (centered on the mean) such that it contains 95% probability for the annual profit of the company.

B.	Specify the 5th percentile of profit (in Rupees) for the company

C.	Which of the two divisions has a larger probability of making a loss in a given year?


In [1]:
import numpy as np
import pandas as pd

# Set the random seed for reproducibility
np.random.seed(42)

# Define the parameters of the normal distributions
mean_profit1 = 5
std_dev_profit1 = np.sqrt(32)
mean_profit2 = 7
std_dev_profit2 = np.sqrt(42)

# Generate random samples for Profit1 and Profit2
num_samples = 10_000  # Number of samples to generate
profit1 = np.random.normal(mean_profit1, std_dev_profit1, num_samples)
profit2 = np.random.normal(mean_profit2, std_dev_profit2, num_samples)

# Convert profits to Rupees using the exchange rate $1 = Rs. 45
exchange_rate = 45
profit1_rupees = profit1 * exchange_rate
profit2_rupees = profit2 * exchange_rate

# Calculate the total profit in Rupees
total_profit_rupees = profit1_rupees + profit2_rupees

# A. Specify a Rupee range (centered on the mean) such that it contains 95% probability for the annual profit of the company.
mean_total_profit_rupees = total_profit_rupees.mean()
std_dev_total_profit_rupees = total_profit_rupees.std()

confidence_interval = 0.95
z_score = 1.96  # For a 95% confidence interval

lower_range = mean_total_profit_rupees - (z_score * std_dev_total_profit_rupees)
upper_range = mean_total_profit_rupees + (z_score * std_dev_total_profit_rupees)

print(f"A. Rupee range (centered on the mean) containing 95% probability: ({lower_range:.2f} Rs, {upper_range:.2f} Rs)")

# B. Specify the 5th percentile of profit (in Rupees) for the company
percentile_5 = np.percentile(total_profit_rupees, 5)
print(f"B. 5th percentile of profit: {percentile_5:.2f} Rs")

# C. Which of the two divisions has a larger probability of making a loss in a given year?
prob_loss_division1 = (profit1 < 0).mean()
prob_loss_division2 = (profit2 < 0).mean()

if prob_loss_division1 > prob_loss_division2:
    division_with_higher_prob_loss = "Division 1"
elif prob_loss_division2 > prob_loss_division1:
    division_with_higher_prob_loss = "Division 2"
else:
    division_with_higher_prob_loss = "Both divisions have the same probability of making a loss"

print(f"C. {division_with_higher_prob_loss} has a larger probability of making a loss in a given year.")


A. Rupee range (centered on the mean) containing 95% probability: (-213.63 Rs, 1300.43 Rs)
B. 5th percentile of profit: -81.58 Rs
C. Division 1 has a larger probability of making a loss in a given year.
